testing-daml-models/daml-scripts/3_queries_and_assertions.md

```diff
@@ -0,0 +1,50 @@
+Just as in scenarios, with DAML script you can issue queries against the ledger and make assertions
+on the results.
+
+Let's query the ledger for `User` contracts from the operator point of view. Paste the following
+line before the final `debug` line.
+
+<pre class="file" data-target="clipboard">
+    users <- query @User operator
+</pre>
+
+1. `query` is parameterized over the template type you are looking for. In this case we're
+   interested in `User` contracts.
+1. `query` takes a party as an argument. It only returns active contracts that are visible to this party.
+1. The returned `users` variable is a list of `(ContractId, User)` tuples, i.e. the contract ID and the
+   contract data for it.
+
+To write assertions you need to import the `DA.Assert` module. We also import the `sort` function
+that we will be using later. Replace 
+
+<pre>
+module User where
+</pre>
+
+with
+
+<pre class="file" data-target="clipboard">
+module User where
+
+import Daml.Script
+import DA.Assert ((===))
+import DA.List (sort)
+</pre>
```

Comment:
Is `import DA.Foldable (forA_)`missing here? IDE was complaining without it

Reply:
yes it was, fixed.